### PR TITLE
[poc] Added React Table Component

### PR DESCRIPTION
**Pull request by thealmightyone**

### CODE REVIEW

**Comment by stepek** [comment]
Please remove console.logs

File: src/components/Table.jsx [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+  handleSubmit = data => {
+    this.props.createData(this.props.schema.url, data, this.props.schema.singular);
+  };
+  handlePageClick() {
+
+  }
+  render() {
+    const {schema, singular} = this.props.schema;
+
+    if (this.props.isLoading || !Array.isArray(this.props.data)) {
```
Comment (stepek) [inline src/components/Table.jsx]: Please remove 
``` !Array.isArray(this.props.data) ```

File: src/components/Table.jsx [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+          style={{margin: '8px 8px 8px 8px', marginRight: 'auto'}}
+          primary={true}
+        />
+        <Dialog open={this.state.openModal} action={this.state.actionModal}
+          onRequestClose={this.handleCloseModal} schema={this.props.schema}
+          onSubmit={this.handleSubmit}
+        />
+        <Table style={{tableLayout: 'auto'}} className="table">
+          <TableBody displayRowCheckbox={false} showRowHover={true}>
+            <TableHeaderComponent schema={schema}/>
```
Comment (stepek) [inline src/components/Table.jsx]: Please move TableHeaderComponent to TableHeader

File: package.json [hunk 3]
```diff
@@ -79,12 +79,14 @@
   },
   "dependencies": {
     "axios": "^0.15.2",
-    "material-ui": "^0.16.0",
-    "react": "^15.3.0",
-    "react-dom": "^15.3.0",
+    "material-ui": "^0.16.3",
+    "materialize-css": "^0.97.8",
```
Comment (stepek) [inline package.json]: why you add 2nd Framework for material ui?

File: src/components/Table.jsx [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+
+function mapStateToProps() {
+  return {};
+}
+
+TableComponent.propTypes = {
+  schema: PropTypes.object.isRequired,
+};
+
+export default connect(mapStateToProps, {
```
Comment (stepek) [inline src/components/Table.jsx]: Please remove connect method call from this module!

File: src/components/TableHeaderComponent.jsx [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+} from 'material-ui';
+
+const columnStyle = {
+  wordWrap: 'break-word',
+  whiteSpace: 'normal'
+};
+
+class TableHeaderComponent extends Component {
+
+  constructor(props) {
```
Comment (stepek) [inline src/components/TableHeaderComponent.jsx]: you don't need override constructor for call only super constructor

File: src/components/TableHeaderComponent.jsx [hunk 6]
```diff
@@ -0,0 +1,53 @@
+import React, {Component, PropTypes} from 'react';
+
```
Comment (stepek) [inline src/components/TableHeaderComponent.jsx]: Please remove this empty line

File: src/components/TableHeaderComponent.jsx [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+};
+
+class TableHeaderComponent extends Component {
+
+  constructor(props) {
+    super(props);
+  }
+
+  render() {
+    const schema = this.props.schema;
```
Comment (stepek) [inline src/components/TableHeaderComponent.jsx]: add new empty line after const declaration.

File: src/components/TableHeaderComponent.jsx [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+            return null;
+          }
+          return (
+            <TableHeaderColumn key={index}
+              tooltip={property.description}
+              style={columnStyle}>{property.title}
+            </TableHeaderColumn>
+          );
+        })}
+        <TableHeaderColumn >{'options'}</TableHeaderColumn>
```
Comment (stepek) [inline src/components/TableHeaderComponent.jsx]: Options

File: src/components/TableRowComponent.jsx [hunk 9]
```diff
@@ -0,0 +1,80 @@
...
+} from 'material-ui';
+
+const columnStyle = {
+  wordWrap: 'break-word',
+  whiteSpace: 'normal'
+};
+
+class TableRowComponent extends Component {
+
+  constructor(props) {
```
Comment (stepek) [inline src/components/TableRowComponent.jsx]: you don't need override constructor for call only super constructor

File: src/components/TableRowComponent.jsx [hunk 10]
```diff
@@ -0,0 +1,80 @@
...
+    super(props);
+  }
+
+  render() {
+    const {schema, singular} = this.props.schema;
+    const {schema: schemaDrop, item, ...propsForChildren} = this.props; // eslint-disable-line no-unused-vars
+    console.log(propsForChildren);
```
Comment (stepek) [inline src/components/TableRowComponent.jsx]: remove...

File: src/components/TableRowComponent.jsx [hunk 11]
```diff
@@ -0,0 +1,80 @@
...
+            iconStyle={{color: 'rgba(0, 0, 0, 0.55)'}}>
+            <EditIcon />
+          </IconButton>
+          <IconButton tooltip="delete" tooltipPosition="bottom-right"
+            iconStyle={{color: 'rgba(0, 0, 0, 0.55)'}}>
+            <DeleteIcon />
+          </IconButton>
+        </TableRowColumn>
+      </TableRow>
+
```
Comment (stepek) [inline src/components/TableRowComponent.jsx]: Please remove empty line.

File: src/index.js [hunk 12]
```diff
@@ -13,16 +13,40 @@ import {fetchConfig} from './config/ConfigActions';
 
 import {updateLocation} from './routes/LocationActions';
 
+import '../css/sass/main.scss';
+
+import 'materialize-css/js/materialize.js';
```
Comment (stepek) [inline src/index.js]: Why you add 2nd material framework ?!

File: src/components/Table.jsx (hunk 2 above)
Comment (thealmightyone) [inline src/components/Table.jsx]: I'm having trouble aligning header columns with body columns when it's moved to TableHeader

File: src/components/Table.jsx [hunk 13]
```diff
@@ -0,0 +1,100 @@
...
+
+const detailStyle = {
+  padding: 15,
+  margin: 15
+};
+const loadingIndicatorStyle = {
+  margin: 'auto',
+  position: 'relative'
+};
+import PaginationComponent from 'react-ultimate-pagination-material-ui';
```
Comment (stepek) [inline src/components/Table.jsx]: WTF?!

Comment (thealmightyone) [inline src/components/Table.jsx]: my bad, didnt update package.json

File: package.json [hunk 14]
```diff
@@ -79,14 +79,16 @@
...
+    "material-ui": "^0.16.3",
     "react": "^15.3.0",
-    "react-dom": "^15.3.0",
+    "react-dom": "^15.2.1",
     "react-jsonschema-form": "^0.40.0",
     "react-modal": "^1.4.0",
     "react-redux": "^4.4.5",
     "react-router": "^2.6.1",
     "react-spinkit": "^1.1.11",
+    "react-tap-event-plugin": "^2.0.0",
```
Comment (stepek) [inline package.json]: Please remove "react-tap-event-plugin" from dev-dependency

Comment (thealmightyone) [inline package.json]: fixed

**Pull request merged**